### PR TITLE
Typed dict validation

### PR DESCRIPTION
**Pull request by dpeng817**
Method for validating a passed in typed dict. This is useful for anomaly detection stuff.

### CODE REVIEW

**Comment by dpeng817** [comment]
* **#20248** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20248?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dpeng817 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by sryza — COMMENTED** [review]
I wonder if this should be inside `dagster._check`?

**Comment by sryza** [comment]
See question above

**Comment by dpeng817** [comment]
ah whoops forgot to re-push. I did end up moving it @sryza

File: python_modules/dagster/dagster/_check/__init__.py [hunk 1]
```diff
@@ -1781,3 +1784,66 @@ def _check_two_dim_mapping_entries(
...
+        elif get_origin(value) is dict:
+            result[key] = {}
+        elif get_origin(value) is NotRequired:
+            continue
+        else:
+            raise Exception("fields must be either optional or typed dicts")
+    return cast(_TypedDictClass, result)
+
+
+def validate_typeddict(cls: Type[_TypedDictClass], potential_typeddict: dict) -> _TypedDictClass:
```
Comment (sryza) [inline python_modules/dagster/dagster/_check/__init__.py:1818]: I think we should switch the arg order here, for consistency with `isinstance` and `check.inst`

**Review by sryza — CHANGES_REQUESTED** [review]
I wonder if it would make sense for `check.inst` to handle this behavior.

```python
check.inst(my_dict, MyTypedDictClass)
```

Not strongly held though. I requested changes only based on my comment about argument order.

**Comment by dpeng817** [comment]
I think that's somewhat of a misuse of the terminology, the passed-in dict isn't an "instance" of the typed dict in the traditional sense, it just matches the spec, so I'd prefer to just keep the methods separate/explicit about what they're doing.

**Comment by dpeng817** [comment]
no longer necessary - switched to pydantic basemodels.

**Pull request closed without merging**